lib/cli/etcher.js

```diff
@@ -18,61 +18,115 @@
+const imageStream = require('etcher-image-stream');
 const writer = require('./writer');
 const utils = require('./utils');
 const options = require('./cli');
+const DynamicList = require('./dynamic-list');
+const DriveScanner = require('./drive-scanner');
 const EXIT_CODES = require('../src/exit-codes');
 
-isElevated().then((elevated) => {
+inquirer.registerPrompt('dynamicList', DynamicList);
+
+const scanner = new DriveScanner(2000);
+
+const driveToChoice = (drive, minSizeToEnable) => {
+  const size = drive.size / 1000000000;
+
+  let displayName = `${drive.device} (${size.toFixed(1)} GB) - ${drive.description}`;
+
+  if (drive.system) {
+    displayName += chalk.red(' FIXED DRIVE');
+  }
+
+  const choice = {
+    name: displayName,
+    value: drive.device
+  };
+  if (drive.size < minSizeToEnable) {
+    choice.disabled = 'TOO SMALL FOR IMAGE';
+  }
+  return choice;
+};
+
+const mustConfirm = !options.robot && !options.yes;
+
+Bluebird.all([
+  isElevated(),
+  imageStream.getImageMetadata(options._[0])
+]).then((results) => {
+  const elevated = results[0];
+  const imageMetadata = results[1];
+
   if (!elevated) {
     throw new Error('This should should be run with root/administrator permissions');
   }
+  const questions = [];
 
-  return form.run([
-    {
+  if (!options.drive) {
+    questions.push({
       message: 'Select drive',
-      type: 'drive',
-      name: 'drive'
-    },
-    {
-      message: 'This will erase the selected drive. Are you sure?',
-      type: 'confirm',
-      name: 'yes',
-      default: false
-    }
-  ], {
-    override: {
-      drive: options.drive,
+      type: 'dynamicList',
+      choices: [],
+      emitter: scanner,
+      name: 'drive',
+      mapFn: (drive) => {
+        return driveToChoice(drive, imageMetadata.size);
+      }
+    });
+  }
 
-      // If `options.yes` is `false`, pass `undefined`,
-      // otherwise the question will not be asked because
-      // `false` is a defined value.
-      yes: options.robot || options.yes || undefined
+  if (mustConfirm) {
+    questions.push({
+      type: 'confirm',
+      name: 'deleteDrive',
+      message: 'This will erase the selected drive. Are you sure?',
+      default: 'n'
+    });
+  }
 
-    }
-  });
+  return inquirer.prompt(questions);
 }).then((answers) => {
-  if (!answers.yes) {
-    throw new Error('Aborted');
+  if (!mustConfirm || answers.deleteDrive) {
+    return answers.drive;
   }
-
-  const progressBars = {
-    write: new visuals.Progress('Flashing'),
-    check: new visuals.Progress('Validating')
-  };
-
+  throw new Error('Aborted');
+}).then((drive) => {
   return drivelist.listAsync().then((drives) => {
+
     const selectedDrive = _.find(drives, {
-      device: answers.drive
+      device: drive
     });
 
     if (!selectedDrive) {
-      throw new Error(`Drive not found: ${answers.drive}`);
+      throw new Error(`Drive not found: ${drive}`);
+    }
+
+    if (selectedDrive.system && mustConfirm) {
```

Comment:
Oops, my bad. `SelectionStateModel.isSystemDrive()` was currently in a branch. I got merged very recently (https://github.com/resin-io/etcher/pull/978), so you can rebase to get it.